graph/graph.go

```diff
@@ -216,18 +217,22 @@ func (g *Graph) InvertEdge(eID string) {
 	}
 }
 
-// removeNodeByID removes the Node with the ID
-func (g *Graph) removeNodeByID(ID string) {
+// RemoveNodeByID removes the Node with the ID and the Edges
+// associated with it
+func (g *Graph) RemoveNodeByID(ID string) error {
```

Comment:
This will now return an error an empty graph, is it expected? I don't think we have situation's like this but that's new